### PR TITLE
Added Post groups API in group list handlers

### PR DESCRIPTION
**Pull request by sangramga**
Solves issue #1602 
Its possible to add/post multiple groups using the post method of "/groups" API.

### CODE REVIEW

File: jupyterhub/tests/test_api.py [hunk 1]
```diff
@@ -966,6 +966,42 @@ def test_groups_list(app):
...
+    assert r.status_code == 201
+    reply = r.json()
+    r_names = [group['name'] for group in reply]
+    assert names == r_names
+
+    # for name in names:
+    #     user = find_user(db, name)
+    #     assert user is not None
+    #     assert user.name == name
+    #     assert not user.admin
```
Comment (minrk) [inline jupyterhub/tests/test_api.py]: Do you want to restore or delete these commented blocks?

Comment (sangramga) [reply jupyterhub/tests/test_api.py]: Oh sorry. I will delete those comments.

**Comment by minrk** [comment]
Thanks!

**Comment by sangramga** [comment]
@minrk Also need to make changes to REST API Documentation(Swagger) to include the above features.

**Pull request merged**